src/@utils/accessDetailsAndPricing.ts

```diff
@@ -229,12 +89,8 @@ export async function getOrderPriceAndFees(
   orderPriceAndFee.providerFee = providerFees || initializeData.providerFee
 
   // fetch price and swap fees
-  if (asset?.accessDetails?.type === 'fixed') {
-    const fixed = await getFixedBuyPrice(
-      asset?.accessDetails,
-      asset?.chainId,
-      signer
-    )
+  if (accessDetails.type === 'fixed') {
```

Comment:
Do you think that can we have an enum instead of fixed, free ecc...?

Reply:
accessDetails are strongly typed with typescript. I would say its enough. `type` is of type `'fixed' | 'free'`It won't let you compare with something else.